samples/snippets/src/main/java/com/example/spanner/HubbleTransactionsCodeLab.java

```diff
@@ -290,6 +242,175 @@ public void updatesAndReads(
         NUM_PROCESSORS / 2);
   }
 
+  public void doWorkSingleTransactionParallelLocking(DatabaseClient dbClient) {
+    // getting the available processor
+    ExecutorService executorService = Executors.newFixedThreadPool(NUM_PROCESSORS);
+
+    List<Future<Integer>> futures = new ArrayList<>();
+    // List to store the count of total number of workItem(row) which has not yet completed means
+    // is_done is false.
+    List<String> workItemCount = new ArrayList<>();
```

Comment:
We don't need this. Instead, we can just inspect the resultSet returned from the executed query. If there are zero rows in it, we know we've done all the work.

Reply:
Here do ups mean result.getRows()--> I did not find this method in the API? or something else

Reply:
result.getRows().size() != 0